src/domain/harbors/HarborsPageContainer.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { useQuery } from '@apollo/react-hooks';
+import { useTranslation } from 'react-i18next';
+
+import { HARBORS_QUERY } from './harborsQuery';
+import { HARBORS } from './__generated__/HARBORS';
+import PropertyIcon from './propertyIcon/PropertyIcon';
+import Table from '../../common/table/Table';
+
+const HarborsContainer: React.FC = () => {
+  const { loading, error, data } = useQuery<HARBORS>(HARBORS_QUERY);
+  const { t } = useTranslation();
+
+  const columns = [
+    {
+      Header: t('harbors.tableHeaders.harbor'),
+      accessor: 'name',
+    },
+    {
+      Header: t('harbors.tableHeaders.places'),
+      accessor: 'numberOfPlaces',
+    },
+    {
+      Cell: ({ cell }: { cell: { value: number } }) => (
+        <PropertyIcon name="plug" disabled={!!cell.value} />
+      ),
+      Header: () => <PropertyIcon name="plug" />,
+      accessor: 'electricity',
+    },
+    {
+      Cell: ({ cell }: { cell: { value: number } }) => (
+        <PropertyIcon name="fence" disabled={!!cell.value} />
+      ),
+      Header: () => <PropertyIcon name="fence" />,
+      accessor: 'gate',
+    },
+    {
+      Cell: ({ cell }: { cell: { value: number } }) => (
+        <PropertyIcon name="streetLight" disabled={!!cell.value} />
+      ),
+      Header: () => <PropertyIcon name="streetLight" />,
+      accessor: 'lighting',
+    },
+    {
+      Cell: ({ cell }: { cell: { value: number } }) => (
+        <PropertyIcon name="waterTap" disabled={!!cell.value} />
+      ),
+      Header: () => <PropertyIcon name="waterTap" />,
+      accessor: 'water',
+    },
+    {
+      Cell: ({ cell }: { cell: { value: number } }) => (
+        <PropertyIcon name="trash" disabled={!!cell.value} />
+      ),
+      Header: () => <PropertyIcon name="trash" />,
+      accessor: 'wasteCollection',
+    },
+  ];
+
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>Error</p>;
+
+  const tableData =
+    data && data.harbors
+      ? data.harbors.edges.reduce<object[]>((acc, harbor) => {
+          if (harbor && harbor.node && harbor.node.properties) {
+            const properties = harbor.node.properties.piers.edges.reduce<{
+              electricity: number;
+              gate: number;
+              lighting: number;
+              wasteCollection: number;
+              water: number;
+            }>(
+              (prev, pier) => {
+                if (pier && pier.node && pier.node.properties) {
```

Comment:
lodash get?

Reply:
I'm a bit torn on this. Lodash would improve readability of this, but I would prefer not adding an extra library for this pattern.

Reply:
Well you might somehow injected lodash by some package dependencies (check yarn.lock). And also partial import doesnt hurt as much as full lib import.

Anyway, bad thing about lodash -> swallowed error.

I was 50:50 about using lodash so no is fine

Reply:
I opted to not use `lodash`. Specifically for this particular issue since optional chaining is in [beta](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/)  now.